extensions/sql/transfer-process-store/sql_tps_proposal.md

```diff
@@ -0,0 +1,133 @@
+## Table schema
+
+```sql
+create table edc.data_request
+(
+    id                varchar                                            not null
+        constraint data_request_pk
+            primary key,
+    process_id        varchar                                            not null,
+    connector_address varchar                                            not null,
+    protocol          varchar default 'ids-multipart'::character varying not null,
+    connector_id      varchar,
+    asset_id          varchar                                            not null,
+    contract_id       varchar                                            not null,
+    data_destination  varchar                                            not null,
+    managed_resources boolean default true,
+    properties        varchar,
+    transfer_type     varchar
+);
+
+comment on column edc.data_request.data_destination is 'DataAddress serialized as JSON';
+
+comment on column edc.data_request.properties is 'java Map serialized as JSON';
+
+comment on column edc.data_request.transfer_type is 'TransferType serialized as JSON';
+
+alter table edc.data_request
+    owner to "user";
+
+create table edc.transfer_process
+(
+    id                       varchar           not null
+        constraint transfer_process_pk
+            primary key,
+    type                     integer default 0 not null,
+    state                    integer           not null,
+    state_count              integer default 0 not null,
+    state_time_stamp         timestamp,
+    trace_context            varchar,
+    error_detail             varchar,
+    resource_manifest        varchar,
+    provisioned_resource_set varchar,
+    data_request_id          varchar           not null
+        constraint data_request___fk
+            references edc.data_request
+            on update restrict on delete restrict
+);
+
+comment on column edc.transfer_process.trace_context is 'Java Map serialized as JSON';
+
+comment on column edc.transfer_process.resource_manifest is 'java ResourceManifest serialized as JSON';
+
+comment on column edc.transfer_process.provisioned_resource_set is 'ProvisionedResourceSet serialized as JSON';
```

Comment:
As all three objects are pretty simple (just `Strings` in a `HashMap` or `String` / `Boolean` as field) I would do the extra effort create dedicated tables for them. This also might become handy when the EDC wants to query them or migrate the schema to another version.

<sub>Dominik Pinsel <dominik.pinsel@daimler.com>, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md) </sub>

Reply:
my reasoning behind serializing them was that to my knowledge there isn't any plans to query on those fields, so it seemed to be the lowest-hanging fruit, because it makes `insert` and `delete` statements simpler.